src/main/webapp/assignPanel.js

```diff
@@ -0,0 +1,266 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/* eslint-disable no-unused-vars */
+/* global getTaskListsAndTasks, postNewTaskList,
+ enableAssignStartResetButton, createTextListElementFromString,
+ encodeListForUrl, AuthenticationError, Task, postNewTask,
+ disableAssignAcceptRejectButtons */
+
+// Script for handling the behaviour of the Assign panel's features
+
+const assignTaskListTitle = 'Actionable Emails (Blueprint)';
+let assignTaskListId;
+let assignTasks = [];
+let nDays;
+let unreadOnly;
+let subjectLinePhrases = [];
+let actionableEmails = [];
+
+const taskNotesPrefix =
+    '*** ' +
+    'Made automatically with Blueprint. ' +
+    'Please add notes above this line, ' +
+    'and do not edit anything below this line. ' +
+    'Blueprint needs this information to better serve you. ' +
+    'Thank you for your understanding. ' +
+    '***' +
+    '\n';
+
+/**
+ * Process for initializing the assign panel after login or a change in settings
+ *
+ * @return {Promise<null>} promise that resolves when all assign global
+ *     variables are initialized
+ */
+function initializeAssignFeatures() {
+  const actionItemsCountElement =
+      document.getElementById('assign-suspected-action-items');
+
+  return Promise.all(
+      [
+        getAssignPanelTasks(),
+        fetchActionableEmails(subjectLinePhrases, unreadOnly, nDays),
+      ]
+  ).then((values) => {
+    // Parse values from server
+    const assignTasksResponse = values[0];
+    const actionableEmailsResponse = values[1];
+
+    assignTaskListId = assignTasksResponse.taskListId;
+    assignTasks = assignTasksResponse.tasks;
+
+    const tasksEmailIds =
+        new Set(assignTasks
+            .map((task) => parseEmailIdFromTaskNotes(task))
+            .filter((id) => id !== null));
+
+    actionableEmails =
+        actionableEmailsResponse
+            .filter((message) => !tasksEmailIds.has(message.id));
+
+    // Set the panel statistics
+    actionItemsCountElement.innerText = actionableEmails.length;
+  });
+}
+
+/**
+ * Will get the email ID from the task notes.
+ * Assumes that the email ID will be in the last line of the notes.
+ *
+ * @param {Task} task task object
+ * @return {string|null} id if present, null otherwise
+ */
+function parseEmailIdFromTaskNotes(task) {
+  // Get emailId from end of message
+  const notes = task.notes;
+  const minimumLength = 'emailId:'.length;
+  if (notes == null || notes.length < minimumLength) {
```

Comment:
nit: you could remove the `notes.length < minimumLength` check

Reply:
Agreed! Thanks!